src/App.tsx

```diff
@@ -1,20 +1,75 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
+import { MovieData } from './types/MovieData';
+import { ResponseError } from './types/ReponseError';
+
+import { getMovie } from './api';
+
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
+  const [showError, setShowError] = useState<boolean>(false);
+
+  const isMovieData = (data: MovieData | ResponseError): data is MovieData => {
+    return (data as MovieData).Title !== undefined;
+  };
+
+  useEffect(() => {
+    if (isSubmitted && query.length) {
+      getMovie(query)
+        .then((movieFromServer) => {
+          if (isMovieData(movieFromServer)) {
+            setMovie({
+              title: movieFromServer.Title,
+              description: movieFromServer.Plot,
+              imgUrl:
+                movieFromServer.Poster !== 'N/A'
+                  ? movieFromServer.Poster
+                  : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+              imdbUrl: `https://www.imdb.com/title/${movieFromServer.imdbID}`,
+              imdbId: movieFromServer.imdbID,
+            });
+            setShowError(false);
+          } else {
+            setShowError(true);
+          }
+        })
+        .catch(() => {
+          setShowError(true);
+        })
+        .finally(() => {
+          setIsSubmitted(false);
+        });
+    }
+  }, [isSubmitted]);
 
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={movies} />
+        {movies.length > 0 && (
```

Comment:
You can use logical operators instead 
```suggestion
        {!!movies.length && (
```